decoder/pokemon.go

```diff
@@ -595,54 +582,31 @@ const SeenType_LureEncounter string = "lure_encounter" // Pokemon has been encou
 // information held.
 // db - the database connection to be used
 // wildPokemon - the Pogo Proto to be decoded
-// spawnId - the spawn id the Pokemon was seen at
 // timestampMs - the timestamp to be used for calculations
-// timestampAccurate - whether the timestamp is considered accurate (eg came from a GMO), and so can be used to create
-// a new exact spawnpoint record
-func (pokemon *Pokemon) updateSpawnpointInfo(ctx context.Context, db db.DbDetails, wildPokemon *pogo.WildPokemonProto, spawnId int64, timestampMs int64, timestampAccurate bool) {
-	if wildPokemon.TimeTillHiddenMs <= 90000 && wildPokemon.TimeTillHiddenMs > 0 {
-		expireTimeStamp := (timestampMs + int64(wildPokemon.TimeTillHiddenMs)) / 1000
-		pokemon.ExpireTimestamp = null.IntFrom(expireTimeStamp)
-		pokemon.ExpireTimestampVerified = true
+func (pokemon *Pokemon) updateSpawnpointInfo(ctx context.Context, db db.DbDetails, wildPokemon *pogo.WildPokemonProto, timestampMs int64) {
+	spawnId, err := strconv.ParseInt(wildPokemon.SpawnPointId, 16, 64)
+	if err != nil {
+		panic(err)
```

Comment:
panic is good at this point because GIN recovers? :)

Reply:
It was in the original code from early days. Probably should just return to be honest. We aren’t in a gin call here but a goroutine.

Reply:
I wrote the panic because something is terribly wrong if this fails.